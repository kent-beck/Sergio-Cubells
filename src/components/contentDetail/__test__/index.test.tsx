import { render, screen } from "@testing-library/react";
import ContentDetail from "..";
import { Car } from "../../../types";

describe("Renders a ", () => {
  const carItem: Car = {
    title: "Coche",
    euroPrice: "29.870 €",
    image: "someImageUrl",
    description: "someDescription",
  };

  test("Content Detail", () => {
    const tree = render(<ContentDetail car={carItem} />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  test("euro price text", () => {
    render(<ContentDetail car={carItem} />);

    const euroPriceTextElement = screen.getByText(/29.870 €/i);

    expect(euroPriceTextElement).toBeInTheDocument();
  });

  test("description text", () => {
    render(<ContentDetail car={carItem} />);

    const descriptionTextElement = screen.getByText(/someDescription/i);

    expect(descriptionTextElement).toBeInTheDocument();
  });

  test("from text", () => {
    render(<ContentDetail car={carItem} />);

    const desdeTextElement = screen.getByText(/desde/i);

    expect(desdeTextElement).toBeInTheDocument();
  });

  test("alt image", () => {
    render(<ContentDetail car={carItem} />);

    const testImage = document.querySelector("img") as HTMLImageElement;

    expect(testImage.alt).toContain("Image for Coche");
  });

  test("src image", () => {
    render(<ContentDetail car={carItem} />);

    const testImage = document.querySelector("img") as HTMLImageElement;

    expect(testImage.src).toContain("http://localhost/someImageUrl");
  });
});
