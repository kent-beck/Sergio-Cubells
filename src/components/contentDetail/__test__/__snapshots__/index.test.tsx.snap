// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a  Content Detail 1`] = `
<DocumentFragment>
  <main
    class="detailContentContainer"
  >
    <img
      alt="Image for Coche"
      class="detailMainImage"
      src="someImageUrl"
    />
    <textarea
      class="detailDescriptionTextArea"
    >
      someDescription
    </textarea>
    <div
      class="detailPriceContent"
    >
      <div
        class="detailPriceFrom"
      >
        desde
      </div>
      <div
        class="detailPrice"
      >
        29.870 â‚¬
      </div>
    </div>
  </main>
</DocumentFragment>
`;
